File: AndroidDatabase/RoomDatabaseUpgrade/app/src/main/java/com/haoyudu/roomdatabaseupgrade/WordDao.java
package com.haoyudu.roomdatabaseupgrade;

import androidx.lifecycle.LiveData;
import androidx.room.*;

import java.util.List;

@Dao
public interface WordDao {
    @Insert
    void insertWords(Word...word);

    @Delete
    void deleteWords(Word...word);

    @Update
    void updateWords(Word...word);

    @Query("DELETE FROM WORD")
    void deleteAllWords();

    @Query("SELECT * FROM WORD ORDER BY ID DESC")
    LiveData<List<Word>> queryWords();



}

File: KotlinTest/KotlinDatabase/app/src/main/java/com/haoyudu/kotlindatabase/WordDao.kt
package com.haoyudu.kotlindatabase

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface WordDao {
   @Insert
   fun insertWords(word:Word)

   @Delete
   fun deleteWords(word:Word)

   @Update
   fun updateWords(word:Word)

   @Query("DELETE FROM WORD")
   fun deleteAllWords()

   @Query("SELECT * FROM WORD ORDER BY ID DESC")
   fun queryAllWords():LiveData<List<Word>>
}